### PR TITLE
[FLINK-24481][docs] Translate buffer debloat documenation to chinese

### PR DESCRIPTION
**Pull request by Myracle**
## What is the purpose of the change

*Translate buffer debloat documenation to chinese.*

## Brief change log

  - *Translate related English to Chinese in mem_setup.md, mem_setup_tm.md and network_mem_tuning.md.*

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by Myracle** [comment]
@flinkbot run azure

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:140]: 这里 credit 翻译成信用值不太好理解。credit 是下游发给上游的，告诉上游自己有空间可以接收多少数据的一种机制。建议采用意译，比如”协调数据传输量“

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:152]:
```suggestion
3. 独占 buffer 的目的是提供一个流畅的吞吐量。当一个 buffer 在传输数据时，另一个 buffer 被填充。当吞吐量比较高时，独占 buffer 的数量是定义 Flink 中 in-flight 数据的主要因素。
```

**Review by xintongsong — COMMENTED** [review]
@Myracle, thanks for the translation.
In addition to the inline comments, I have some general suggestions for the translation efforts.
- It's not necessary for the translation to exactly match the original English. As long as it presents the information correctly and precisely, it is more important that the translation itself reads smooth, natural and easy to understand. For that purpose, sometimes you would need to re-write the sentences / paragraphs.
- This [guidelines](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications) provides suggestions on common translation issues and how terms should be translated, and it would be appreciated to help complete the wiki by reporting and discussing issues you meet in translations.

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (Myracle) [reply docs/content.zh/docs/deployment/memory/network_mem_tuning.md:114]: It seems a mistake. Please help to verify the doc. Thank you. @akalash

Comment (akalash) [reply docs/content.zh/docs/deployment/memory/network_mem_tuning.md:114]: @Myracle, @xintongsong, thanks for emphasizing this. It is indeed a mistake. I think the correct way to fix this it is exactly what you proposed - instead of `Like the input buffer pool` it should be `Unlike the input buffer pool`. 
The main idea is that if we have at least one buffer for each subpartition it will be enough to make a progress(It doesn't matter how many buffers we configured) while all exclusive buffers for the input channel have to be allocated on init and if there are less available buffers than we configured, Flink will fail.

@Myracle, can you fix this in English version in a separate commit as well? Or I can do it later.

**Review by xintongsong — COMMENTED** [review]
@Myracle

I haven't got time for a full pass of review. I've only reviewed until the section *Input network buffers*, and there are already some comments which I think it might be better to get them back to you early, rather than wait for another time slot to complete the review.

Overall, I think the translations are much better than the previous round. So far, I don't find anything incorrect or misleading. My comments are mostly on the readability.
- I'd suggest to be cautious with the usage of English terms. When deciding whether a term should be translated or not, an important question is whether translating the term makes it easier for Chineses readers to understand it.
  - In that consideration, I think `in-flight`, `input gate / channel` are better translated.
  - It's probably neutral for `buffer`, as most readers understand it well in English. However, it would be better to stay consistent throughout the document. To be specific, I'd be fine with either `buffer` and `网络 buffer`, or `缓冲` and `网络缓冲`, but not a mixture of them.
- There are quite some formatting issues, like missing or extra spaces. Such things can be easily overlooked in markdown files. It would be helpful to proofread the compiled version of the document.
- Changes to the English content should be in a separate hotfix commit.
- It would be helpful to address comments in separate fix-up commits, so that changes since the last review can be easily identified. Otherwise, the reviewers would have to go through all the changes for every round of the review.

**Comment by Myracle** [comment]
@xintongsong Thank you for the review. I have modified the code. Please review again.

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:54]:
```suggestion
* 有太多缓冲数据对 checkpoint barrier 推进或者非对齐的 checkpoint 的大小造成不良影响。
```

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:82]:
```suggestion
如果您的 subtask 有很多不同的输入或者有一个合并的输入，缓冲消胀可能会导致低吞吐的输入有太多缓冲数据，而高吞吐输入的缓冲区数量可能太少而不够维持当前吞吐。当不同的输入吞吐差别比较大时，这种现象会更加的明显。我们推荐您在测试这个功能时重点关注这种 subtask。
```

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:115]:
```suggestion
不同于输入缓冲区池，这里配置的独占缓冲区和流动缓冲区只被当作推荐值。如果没有足够的缓冲区，每个输出 subpartition 可以只使用一个独占缓冲区而没有流动缓冲区。
```

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:119]:
```suggestion
独占缓冲区和流动缓冲区的默认配置应该足以应对最大吞吐。如果想要最小化缓冲数据量，那么可以将独占缓冲区设置为 `0`，同时减小内存段的大小。
```

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:125]:
```suggestion
如果缓冲区太小（比如小于一条 record），会因为开销比较大而导致吞吐低。
```

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:143]:
```suggestion
使用 buffer 往返时间（大概 `1ms` 在正常的本地网络中），[缓冲区大小]({{< ref "docs/deployment/config" >}}#taskmanager-memory-segment-size)和期待的吞吐，您可以通过下面的公式计算维持吞吐所需要的缓冲区数量：
```

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:147]:
```suggestion
比如，期待吞吐为 `320MB/s`，往返延迟为 `1ms`，内存段为默认大小，为了维持吞吐需要使用10个活跃的缓冲区：
```

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:151]:
```suggestion
2. 流动缓冲区的目的是为了处理数据倾斜。理想情况下，流动缓冲区的数量（默认8个）和每个通道独占缓冲区的数量（默认2个）能够使网络吞吐量饱和。但这并不总是可行和必要的。所有 subtask 中只有一个通道被使用也是非常罕见的。
```

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:153]:
```suggestion
3. 独占缓冲区的目的是提供一个流畅的吞吐量。当一个缓冲区在传输数据时，另一个缓冲区被填充。当吞吐量比较高时，独占缓冲区的数量是决定 Flink 中缓冲数据的主要因素。
```

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:162]:
```suggestion
- 使用默认值以获得最大吞吐
```

File: docs/content.zh/docs/deployment/memory/network_mem_tuning.md
Comment (xintongsong) [inline docs/content.zh/docs/deployment/memory/network_mem_tuning.md:163]:
```suggestion
- 减少内存段大小、独占缓冲区的数量来加快 checkpoint 并减少网络栈消耗的内存量
```

**Review by xintongsong — APPROVED** [review]
Thanks for addressing my comments, @Myracle.

The PR looks quite good to me now. I'm taking over from here. 

There are still some minor comments, which I'll address myself during merging.

**Comment by xintongsong** [comment]
I've pushed the branch with the following changes:
- Reorganized commit history. Hotfix is preferred to be committed before the major changes, and should fix for both English and the untranslated Chineses documents.
- Updated the commit message.
- Addressed the last comments in a fixup commit.

@Myracle, please double-check on the changes I made. I'll merge after your confirmation, unless there's other concerns.

**Comment by Myracle** [comment]
@xintongsong Thanks for the last changes which are really helpful. I have no problem.

**Comment by Myracle** [comment]
@flinkbot  run azure

**Comment by xintongsong** [comment]
@flinkbot run azure

**Comment by xintongsong** [comment]
Merging as the PR passes compiling and doc_404_check in CI. Failures are irrelevant.

**Pull request closed without merging**
